pkgs/data/themes/adwaita-qt/default.nix

```
@@ -30,11 +31,18 @@ mkDerivation rec {
 
   buildInputs = [
     qtbase
-    qt5.qtx11extras
   ] ++ lib.optionals stdenv.isLinux [
     xorg.libxcb
+  ] ++ lib.optionals (!useQt6) [
+    qt5.qtx11extras
+  ] ++ lib.optionals useQt6 [
+    qtwayland
   ];
 
+  dontWrapQtApps = true;
+
+  cmakeFlags = lib.optionals useQt6 [ "-DUSE_QT6=true" ];
```

Please use coding style optimized for diffs in GNOME-adjacent packages:

```suggestion
  cmakeFlags = lib.optionals useQt6 [
    "-DUSE_QT6=true"
  ];
```